detection-rules/attachment_qr_code_suspicious_components.yml

```
@@ -62,18 +66,29 @@ source: |
                     )
   
                     // exclude google maps
-                    and not strings.starts_with(.scan.qr.url.url, 'https://goo.gl/maps')
-                    and not strings.starts_with(.scan.qr.url.url, 'https://maps.app.goo.gl')
+                    and not strings.starts_with(.scan.qr.url.url,
+                                                'https://goo.gl/maps'
+                    )
+                    and not strings.starts_with(.scan.qr.url.url,
+                                                'https://maps.app.goo.gl'
+                    )
                   )
   
                   // the QR code url is a bing open redirect
-                  or .scan.qr.url.domain.root_domain == 'bing.com' and .scan.qr.url.path =~ '/ck/a'
+                  or (
+                    .scan.qr.url.domain.root_domain == 'bing.com'
+                    and .scan.qr.url.path =~ '/ck/a'
+                  )
+                  // QR code contains non ascii chars
+                  or regex.contains(.scan.qr.url.url, '[^\x00-\x7F]')
                   or (
   
                     // usap-dc open redirect
                     .scan.qr.url.domain.root_domain == "usap-dc.org"
                     and .scan.qr.url.path =~ "/tracker"
-                    and strings.starts_with(.scan.qr.url.query_params, "type=dataset&url=http")
+                    and strings.starts_with(.scan.qr.url.query_params,
+                                            "type=dataset&url=http"
+                    )
```

Lets move this out of being limited to where the `.scan.qr.url.domain.root_domain in $url_shorteners` and out as it's own "or" condition.  Example to include base64 scanning

[hunt with this logic](https://platform.sublime.security/messages/hunt?huntId=01955d11-78c6-7e6c-bcea-951dae30de85)

 
```suggestion
                    )
                    or (
                      any(recipients.to,
                          strings.icontains(..scan.qr.url.url, .email.email)
                          or any(beta.scan_base64(..scan.qr.url.url,
                                                  ignore_padding=true
                                 ),
                                 strings.icontains(., ..email.email)
                          )
                          or any(beta.scan_base64(..scan.qr.url.fragment,
                                                  ignore_padding=true
                                 ),
                                 strings.icontains(., ..email.email)
                          )
                      )
                    )
```